### DISCUSSION TITLE
How to start a computation from a previous one ?

### DISCUSSION THREAD

**Discussion by salettid** [damask-multiphysics/DAMASK#35] [Q&A] [answered]
Hello,
I've done a computation with Damask grid. Let's call it compA.
I would like to get the results of compA (up to this, it's ok) and then launch a new computation with an additionnal increment of load that would start from the end of compA without repeating it. 
How can it be handled in Damask. I saw some restart option but not sure on how it can be used.

Thanks.

**Comment by hyharry** [comment]
in your load file, you can specify the interval to write restart with keyword `f_restart`, and you know the last load increment `n_restart` where a restart is written. then you can resume your simulation with the command like this

```
DAMASK_grid -g geom.vti -l load.yaml -m material.yaml -r n_restart
```

the simulation will then resume at `n_restart` and then continue with the added loadsteps in your `load.yaml`

**Reply by salettid** [reply]
Thanks. 
But how does it know from which computation to restart ? Is it looking at the last hdf5 file result ? 

For instance, I'm doing a first computation, in my command line, I write the command :
`DAMASK-grid -g geom.vti -l load.yml -m material.yaml`
In the first load.yaml file, I specify f_restart = 1, for instance.

I get the results, do some post-processes and decide to go a step further, restarting from the last increment. I detect that the last increment is the 200th . I add a loadcase at the end of my  initial load.yaml file and write :

`DAMASK_grid -g geom.vti -l load.yaml -m material.yaml -r 200`   ?

Sorry if I'm not clear enough ...

**Reply by hyharry** [reply, marked as answer]
in `load.yaml` one can specify `f_out` (output frequency for `result.hdf5`) and `f_restart` (restart output frequency which always overwrites a `restart.hdf5`). When restarting, simulation will refer to the last written restart increment `n_restart` (according to your `f_restart`) not the last recording in your `result.hdf5` (according to your `f_out`). 

to determine the last restart increment, you can check your screen output for the last restart statement, or if you redirect this stdout to a log file, then you should be able to search the last restart statement in your log file.

You could also have a look at our [documentation page](https://damask-multiphysics.org/documentation/file_formats/grid_solver.html#loadstep) for further information.

Hope it clarifies your question.

**Comment by salettid** [comment]
Thanks.
I can now restart from the desired increment but I get error message regarding an error in H5Gcreate2() function.
I am using 3.0.0-beta.

By the way, the documentation page is not dealing with an int as arg of -r option.

Thanks again.

**Reply by MarDiehl** [reply]
> By the way, the documentation page is not dealing with an int as arg of -r option.

thanks, will fix that.
